### ISSUE TITLE
Why has changing GPU's mid training reduced my mAP and increased my 'avg loss'?

### ISSUE THREAD

**Issue by aicukltd** [AlexeyAB/darknet#5655]
**Background**
I am training a model on my local development workstation (LDW).

My LDW has 2x nVidia RTX 2080Ti's running in SLI mode.

I also have a cloud VM (CVM) with 4x nVidia Tesla V100's  which I wish to use to train the same model once my LDW hits around 20000 iterations or a mAP of > 80% (whichever comes first).

I have directly mirrored the config, training data, names and labels from my LDW to the CVM and I receive the follow anomalies. 

The only difference between the 2 CMD's is I am telling my LDW to use -gpus 0,1 and the CVM to use -gpus 0,1,2,3.

It is important to note that the config I have based my training CFG on is the config found in the YOLO_MARK repository.

Darknet was compiled separately on each system with the correct MAKEFILE or CMAKE configuration relevant to the capabilities of that system. (ARCH, CUDNN etc)

**Issue**
At 40000 iterations on my LDW I have an 'avg loss' of 0.75~ and a mAP 83%. When I copy the '{datasetName}_40000.weights to the CVM and start training from that point my mAP is 7% and my 'avg loss' on the CVM is between 2.2 and 4.4.

**Questions**
1. Is there a problem with continuing training on a different system with a different GPU configuration?

2. Should the config have changed to match the GPU settings?

3. Have I missed a key fundamental point with the re-commencement of training on a different GPU architecture?

**Comment by AlexeyAB** [comment]
1-3 No.

You are something doing wrong.
I think cfg-file, train dataset, valid dataset or training commad are different between two 1st and 2nd systems.

**Comment by aicukltd** [comment]
From reading your entire documentation set a few times, that is what I thought, but I wanted to ask anyway!

So my config is identical on both systems, it is derived from:
https://github.com/AlexeyAB/Yolo_mark/blob/master/x64/Release/yolo-obj.cfg

The only difference between the config in YOLO MARK and my config's are the **classes** and **filters**. These have been changed to reflect my dateset and names.

On LDW I am training using the following command:
`darknet.exe detector train data/{datasetName}.data cfg/{datasetName}.cfg backup/{datasetName}_last.weights -gpus 0,1 -map`

It is producing the following output:
![chart_dataset](https://user-images.githubusercontent.com/10234035/82206955-0409f580-9901-11ea-8d6c-3557aa895cbb.png)

On CVM I am training using the following command:
`darknet.exe detector train data/{datasetName}.data cfg/{datasetName}.cfg backup/{datasetName}_{latestIteration}.weights -gpus 0,1,2,3 -map`

It is producing the following output:
![chart_cvm](https://user-images.githubusercontent.com/10234035/82207133-42071980-9901-11ea-8d4b-ae0ffbafdfc4.png)

The **only** thing that could be different is the way I copied the (training dataset JPGs) files to the server, I will attempt to mirror that process and report back.

**Comment by AlexeyAB** [comment]
Have you changed even a single character in the cfg file?

If your accuracy increases rapidly, then just wait for the loss and accuracy to stabilize. If your dataset is very small or the leering rate is very high, then this may be due to the zeroing of the optimizer parameters when the training is stopped and restarted (regardless of which system you are running on).

**Comment by aicukltd** [comment]
No, only the 2 lines as instructed in Step 3 of the following repository:
https://github.com/AlexeyAB/Yolo_mark#yolo_mark

Since I first posted, it seems that my 'avg_loss' is stabilising, however the rate at which the mAP is increasing (25-29-31-31.3-33-etc) is decreasing rapidly even though the 'avg_loss' is now around the 1.0 mark!
